test/functional/unattended_controller_test.rb

```diff
@@ -358,6 +358,47 @@ class UnattendedControllerTest < ActionController::TestCase
   end
 
   test 'host template provision URL can be generated from routes' do
-    assert_routing '/unattended/provision', {:controller => 'unattended', :action => 'host_template', :kind => 'provision'}
+    assert_routing '/unattended/provision', {:controller => 'unattended', :action => 'host_template', :kind => 'provision', :format => 'text'}
+  end
+
+  test "should not render a template to anonymous user" do
+    User.current = nil
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}
+    assert_response :redirect
+  end
+
+  test "should not render a template to user w/o email" do
+    User.current = FactoryGirl.create(:user)
+    User.current.auth_source.stubs(:authenticated?).returns(false)
+    session = SETTINGS[:login] ? {:user => User.current.id, :expires_at => 5.minutes.from_now} : {}
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}, session
+    assert_response :unprocessable_entity
+  end
+
+  test "should not render a template to unauthorized user" do
+    User.current = FactoryGirl.create(:user, :with_mail)
+    User.current.auth_source.stubs(:authenticated?).returns(false)
+    session = SETTINGS[:login] ? {:user => User.current.id, :expires_at => 5.minutes.from_now} : {}
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}, session
+    assert_response :forbidden
+    assert_template nil
+  end
+
+  test 'should render a template to user with valid filter' do
+    User.current = FactoryGirl.create(:user, :with_mail, :admin => false)
+    FactoryGirl.create(:filter, :role => User.current.roles.first, :permissions => Permission.where(:name => 'view_hosts'), :search => "name = #{@rh_host.name}")
+    session = SETTINGS[:login] ? {:user => User.current.id, :expires_at => 5.minutes.from_now} : {}
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}, session
+    assert_response :success
+    assert @response.body.include?("linux")
+  end
+
+  test 'should not render a template to user with valid filter' do
+    User.current = FactoryGirl.create(:user, :with_mail, :admin => false)
+    FactoryGirl.create(:filter, :role => User.current.roles.first, :permissions => Permission.where(:name => 'view_hosts'), :search => "name = does_not_exist")
+    session = SETTINGS[:login] ? {:user => User.current.id, :expires_at => 5.minutes.from_now} : {}
```

Comment:
And ditto in all of these tests, the same comments about using the existing `set_session_user`.